Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Main.js

```diff
@@ -222,8 +222,21 @@ function documentReady()
         row.appendChild(cell);
 
         cell = document.createElement("td");
-        var view = new BuildbotBuilderQueueView(platformQueues.builders);
-        cell.appendChild(view.element);
+        var queues = {};
+        var generators = [];
+        platformQueues.builders.forEach(function(queue) {
+            var generator = queue.viewGenerator ? queue.viewGenerator : BuildbotBuilderQueueView;
+            if (!queues[String(generator)]) {
+                queues[String(generator)] = [];
+                generators.push(generator);
+            }
+            queues[String(generator)].push(queue);
+        });
+
+        generators.forEach(function(generator) {
```

Comment:
I think that this deserves a FIXME that creating more than one view per cell is not great, and temporary. I think that it will mostly work save for some UI shortcomings (maybe incorrect spacing between builders, incorrect order within a cell).

Reply:
I'll add a fix me! UI wise, this doesn't end up looking different, you have to enter inspector to see that there are multiple views in a single cell. The order might be off, but only if you mix different queue view types in `platformQueues.builders` (which is possible, I suppose, but would be weird in practice), in cases where the builders are already coupled with their view generator cohort, order will be preserved.